tests/e2e/test_historical_features.py

```diff
@@ -0,0 +1,124 @@
+        name="customer_id", description="Customer", value_type=ValueType.INT64
+    )
+    feast_client.apply_entity(customer_entity)
+
+    max_age = Duration()
+    max_age.FromSeconds(2 * 86400)
+
+    transactions_feature_table = FeatureTable(
+        name="transactions",
+        entities=["customer_id"],
+        features=[
+            Feature("daily_transactions", ValueType.DOUBLE),
+            Feature("total_transactions", ValueType.DOUBLE),
+        ],
+        batch_source=FileSource(
+            "event_timestamp",
+            "created_timestamp",
+            "parquet",
+            os.path.join(staging_path, "transactions"),
+        ),
+        max_age=max_age,
+    )
+
+    feast_client.apply_feature_table(transactions_feature_table)
+
+    retrieval_date = (
+        datetime.utcnow()
+        .replace(hour=0, minute=0, second=0, microsecond=0)
+        .replace(tzinfo=None)
+    )
+    retrieval_outside_max_age_date = retrieval_date + timedelta(1)
+    event_date = retrieval_date - timedelta(2)
+    creation_date = retrieval_date - timedelta(1)
+
+    customers = [1001, 1002, 1003, 1004, 1005]
+    daily_transactions = [np.random.rand() * 10 for _ in customers]
+    total_transactions = [np.random.rand() * 100 for _ in customers]
+
+    transactions_df = pd.DataFrame(
+        {
+            "event_timestamp": [event_date for _ in customers],
+            "created_timestamp": [creation_date for _ in customers],
+            "customer_id": customers,
+            "daily_transactions": daily_transactions,
+            "total_transactions": total_transactions,
+        }
+    )
+
+    feast_client.ingest(transactions_feature_table, transactions_df)
+
+    feature_refs = ["transactions:daily_transactions"]
+
+    customer_df = pd.DataFrame(
+        {
+            "event_timestamp": [retrieval_date for _ in customers]
+            + [retrieval_outside_max_age_date for _ in customers],
+            "customer_id": customers + customers,
+        }
+    )
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        df_export_path = os.path.join(tempdir, "customers.parquets")
+        customer_df.to_parquet(df_export_path)
+        scheme, _, remote_path, _, _, _ = urlparse(staging_path)
+        staging_client = get_staging_client(scheme)
+        staging_client.upload_file(df_export_path, None, remote_path)
+        customer_source = FileSource(
+            "event_timestamp",
+            "event_timestamp",
+            "parquet",
+            os.path.join(staging_path, os.path.basename(df_export_path)),
+        )
+
+        job = feast_client.get_historical_features(feature_refs, customer_source)
+        output_dir = job.get_output_file_uri()
+
+        _, _, joined_df_destination_path, _, _, _ = urlparse(output_dir)
+        joined_df = pd.read_parquet(joined_df_destination_path)
+
+        expected_joined_df = pd.DataFrame(
+            {
+                "event_timestamp": [retrieval_date for _ in customers]
```

Comment:
Are you at all testing the as_of join functionality? Looks like you just have one timestamp in the range?